### PR TITLE
Remove device category from upgrades

### PR DESCRIPTION
**Pull request by danielui**
# Pull Request summary:
- Simplified usage of upgrades by removing DeviceCategory and DeviceType from SoftwareAction and PartitionManager
- Moved common part (InstallSpecification) of SoftwareAction and PartitionManger to utils/upgrades_helper
- Removed DeviceVersionPayload  from set/remove partitions methods. It generates automatically, user passes just DataSequence of devices. Optionally user can pass partition argument, if no - current partiton will be set as default/all available partitions will be removed
 - Add SoftwareAction to api_container
 - add validator to check if all devices have same personality

# Checklist:
- [x] Make sure to run pre-commit before commiting changes
- [x] Make sure all checks have passed
- [x] PR description is clear and comprehensive
- [x] Mentioned the issue that this PR solves (if applicable)
- [x] Make sure you test the changes

### CODE REVIEW

File: vmngclient/utils/upgrades_helper.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    version_type: VersionType
+    device_type: DeviceType
+
+
+class InstallSpecHelper(Enum):
+
+    VMANAGE = InstallSpecification(Family.VMANAGE, VersionType.VMANAGE, DeviceType.VMANAGE)  # type: ignore
+    VSMART = InstallSpecification(Family.VEDGE, VersionType.VMANAGE, DeviceType.CONTROLLER)  # type: ignore
+    VBOND = InstallSpecification(Family.VEDGE, VersionType.VMANAGE, DeviceType.CONTROLLER)  # type: ignore
+    VEDGE = InstallSpecification(Family.VEDGE, VersionType.VMANAGE, DeviceType.VEDGE)  # type: ignore
```
Comment (tehAgitto) [inline vmngclient/utils/upgrades_helper.py:37]: Can we attach some info about why we cannot reuse Personality?

File: vmngclient/api/software_action_api.py [hunk 2]
```diff
@@ -69,30 +28,27 @@ class SoftwareActionAPI:
     session = create_vManageSession(...)
 
     # Prepare devices list
-    devices = [device for device in DevicesAPI(session).devices
-                if device .personality == Personality.VSMART]
+    devices = DevicesAPI(session).get()
```
Comment (tehAgitto) [inline vmngclient/api/software_action_api.py:31]:
```suggestion
    devices = session.api.devices.get()
```

File: vmngclient/api/partition_manager_api.py [hunk 3]
```diff
@@ -30,69 +25,88 @@ class PartitionManagerAPI:
...
+            RemovePartitionPayload(device.deviceId, device.deviceIP, device.version)  # type: ignore
+            for device in payload_devices
         ]
+
         url = "/dataservice/device/action/removepartition"
         payload = {
             "action": "removepartition",
             "devices": [asdict(device) for device in remove_partition_payload],  # type: ignore
-            "deviceType": self.device_type.value,
+            "deviceType": get_install_specification(devices.first()).device_type.value,
```
Comment (tehAgitto) [inline vmngclient/api/partition_manager_api.py:101]: Should be there a validation whether there are not mixed devices type?

**Review by tehAgitto — CHANGES_REQUESTED** [review]

File: vmngclient/api/partition_manager_api.py (hunk 3 above)
Comment (danielui) [reply vmngclient/api/partition_manager_api.py:101]: Quite good idea, I thought about it. I suppose that in most cases, payload would be generated by filter(), in how many cases different personalities would be returned?

File: vmngclient/utils/upgrades_helper.py (hunk 1 above)
Comment (danielui) [reply vmngclient/utils/upgrades_helper.py]: Sure, done.

File: vmngclient/api/partition_manager_api.py (hunk 3 above)
Comment (tehAgitto) [reply vmngclient/api/partition_manager_api.py:101]: We cannot limit users to use only `.filter()` method with our upgrades. 
A simple one-liner would work:
```python
len(set(device.personality for device in devices)) == 1
```
Also we need to add a proper message.
Anything else which is similar would work also.

**Review by tehAgitto — APPROVED** [review]

**Review by sbasan — APPROVED** [review]
lgtm

**Pull request merged**